### PR TITLE
LPS-82215

### PR DESCRIPTION
**Pull request by ChrisKian**
CC @robertodiaz

https://issues.liferay.com/browse/LPS-82215

Added the tests as asked for in https://github.com/shuyangzhou/liferay-portal/pull/6311.

Let me know if you have any questions.  Thanks!

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test

File: portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/DynamicQueryTest.java [hunk 1]
```diff
@@ -69,6 +71,24 @@ public void testCriterion() {
...
+		for (long i =
+			0; i < (PropsValues.DATABASE_IN_MAX_PARAMETERS + 1); i++) {
+
+			parameterList.add(i);
+		}
+
+		dynamicQuery.add(
+			RestrictionsFactoryUtil.in("classNameId", parameterList));
+
+		ClassNameLocalServiceUtil.dynamicQuery(dynamicQuery);
```
Comment (Preston-Crary) [inline portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/DynamicQueryTest.java]: Please assert something in your test here.

File: portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Roberto Díaz
+ */
+public class RestrictionsFactory {
```
Comment (Preston-Crary) [inline portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java]: Please rename to RestrictionsFactoryTest.

File: portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/DynamicQueryTest.java [hunk 3]
```diff
@@ -69,6 +71,24 @@ public void testCriterion() {
 		Assert.assertEquals(className, classNames.get(0));
 	}
 
+	@Test
+	public void testInRestrictionCriterion() {
+		DynamicQuery dynamicQuery = ClassNameLocalServiceUtil.dynamicQuery();
+
+		List<Long> parameterList = new ArrayList();
```
Comment (Preston-Crary) [inline portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/DynamicQueryTest.java]: Please initialize your List's with the size they are going to be if known and use <> operators when they apply.

File: portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		List<Integer> parameterList = new ArrayList();
+
+		for (int i = 0; i < (PropsValues.DATABASE_IN_MAX_PARAMETERS - 1); i++) {
+			parameterList.add(i);
+		}
+
+		Criterion inCriterion = RestrictionsFactoryUtil.in(
+			"property", parameterList);
+
+		Assert.assertFalse(inCriterion instanceof Disjunction);
```
Comment (Preston-Crary) [inline portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java]: Please assert with some information when the assert fails.  Here you could use inCriterion.toString() as the message.

File: portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testInWithLessThanDatabaseInMaxParametersValue() {
+		List<Integer> parameterList = new ArrayList();
+
+		for (int i = 0; i < (PropsValues.DATABASE_IN_MAX_PARAMETERS - 1); i++) {
```
Comment (Preston-Crary) [inline portal-impl/test/integration/com/liferay/portal/dao/orm/hibernate/RestrictionsFactory.java]: We have tests for MAX - 1 and MAX + 1.  Can we do MAX and MAX + 1 instead?

**Comment by Preston-Crary** [comment]
Please see my comments and update the pull with needed changes.  Thanks!

**Pull request closed without merging**